src/common/components/page/DialogmotePage.tsx

```diff
@@ -1,36 +1,90 @@
-import PersonvernInfo from "@/common/components/personvern/PersonvernInfo";
-import React, { ReactNode } from "react";
-import PageHeader from "@/common/components/header/PageHeader";
+import React from "react";
 import Head from "next/head";
+import { People } from "@navikt/ds-icons";
+import PersonvernInfo from "@/common/components/personvern/PersonvernInfo";
+import PageTitle from "@/common/components/header/PageTitle";
 import AppSpinner from "@/common/components/spinner/AppSpinner";
+import { PageContainer } from "@navikt/dinesykmeldte-sidemeny";
+import { Sykmeldt } from "../../../types/shared/sykmeldt";
+import { ArbeidsgiverSideMenu } from "@/common/components/menu/ArbeidsgiverSideMenu";
+import { addSpaceAfterEverySixthCharacter } from "@/common/utils/stringUtils";
+
+type WithNavigationAndHeader =
+  | {
+      sykmeldt: Sykmeldt | undefined;
+      withAGNavigation?: boolean;
+      withAGHeader?: boolean;
+    }
+  | {
+      sykmeldt?: never;
+      withAGNavigation?: never;
+      withAGHeader?: never;
+    };
 
-interface Props {
+type Props = {
   title: string;
-  hideHeader?: boolean;
   isLoading: boolean;
-  children: ReactNode;
-}
+  children: React.ReactNode;
+  hideTitle?: boolean;
+} & WithNavigationAndHeader;
 
 export const DialogmotePage = ({
   title,
-  hideHeader,
   isLoading,
   children,
+  hideTitle,
+  withAGNavigation,
+  withAGHeader,
+  sykmeldt,
 }: Props) => {
   const renderContent = () => {
-    if (isLoading) {
-      return <AppSpinner />;
-    }
+    const sykmeldtName = sykmeldt?.navn ?? "";
+    const sykmeldtFnr = sykmeldt?.fnr ?? "";
+
+    const hasSykmeldtValidData =
+      !!sykmeldtName && !!sykmeldtFnr && !!sykmeldt?.narmestelederId;
+    const showAGHeader = withAGHeader && hasSykmeldtValidData;
```

Comment:
Vi bør ikke ha noe SM eller AG-spesifikk kode her. Dette bør refaktoreres slik at DialogmotePage er helt felles, og så kan man heller sende inn det som er spesifikt for AG/SM.

Reply:
Fikset